projects/hslayers/src/components/add-data/vector/add-data-vector.service.ts

```diff
@@ -276,16 +257,44 @@ export class HsAddDataVectorService {
       console.log('Uploaded file is not supported!');
     }
   }
+
+  async readUploadedFile(file: any): Promise<any> {
+    let uploadedData: any = {};
+    const fileType = this.tryGuessTypeFromNameOrUrl(file.name.toLowerCase());
+    switch (fileType) {
+      case 'kml':
+        uploadedData = await this.createVectorObjectFromXml(file, fileType);
```

Comment:
@FilipLeitner The only problem with this is that You cannot get features array from xml, as they are included inside the base64 url, with the styling.

Reply:
Ye feature count doesnt work with kml but it could be read with kml parser. Should probably add it to not keep it missleading. But this approach is used only when adding KML 'as a whole' . Otherwise, when adding kml features into existing layer 'convertUploadedData' data function is used. Should work.

Reply:
Yes it works. Kinda doesn't feel right parsing data just for the feature count to pop up, but it must be done.

Reply:
Ye thats why I havent done it right away. I could maybe just change description for KML to not include the number. What do you think?

Reply:
Agree on this, but lets also hear what others think of this idea.

Reply:
I think its ok to parse before adding the layer. This way we might catch parsing errors before some broken layer gets added to map.